test/recording-history.js

```diff
@@ -0,0 +1,300 @@
+const Sentry = mockImport('./utils/sentry', 'sentry');
+
+const {shell} = require('electron');
+
+const {
+  recordingHistory,
+  getPastRecordings,
+  checkForActiveRecording,
+  addRecording,
+  setCurrentRecording,
+  updatePluginState,
+  stopCurrentRecording,
+  cleanPastRecordings
+} = require('../main/recording-history');
+
+const incomplete = path.resolve(__dirname, 'fixtures', 'incomplete.mp4');
+const corrupt = path.resolve(__dirname, 'fixtures', 'corrupt.mp4');
+
+test.before(t => {
+  t.context.now = new Date('2020-07-21T15:27:26.564Z');
+  t.context.clock = sinon.useFakeTimers(t.context.now.getTime());
+});
+
+test.after(t => {
+  if (t.context.clock) {
+    t.context.clock.restore();
+  }
+});
+
+test.beforeEach(() => {
+  recordingHistory.clear();
+});
+
+test.afterEach.always(t => {
+  if (t.context.paths) {
+    for (const path of t.context.paths) {
+      if (fs.existsSync(path)) {
+        fs.unlinkSync(path);
+      }
+    }
+  }
+});
+
+test('getPastRecordings', t => {
+  const existingPath = tempy.file({extension: 'mp4'});
+  const missingPath = tempy.file({extension: 'mp4'});
+
+  fs.writeFileSync(existingPath, 'data');
+  t.context.paths = [existingPath];
+
+  recordingHistory.set('recordings', [{filePath: existingPath}, {filePath: missingPath}]);
+
+  t.deepEqual(getPastRecordings(), [{filePath: existingPath}]);
+  t.deepEqual(recordingHistory.get('recordings'), [{filePath: existingPath}]);
+});
+
+test('checkForActiveRecording with no recording', async t => {
+  t.false(await checkForActiveRecording());
+});
+
+test('checkForActiveRecording with playable recording', async t => {
+  const fakeService = {
+    title: 'Fake Service',
+    cleanUp: sinon.fake()
+  };
+
+  const fakePlugin = {
+    name: 'kap-fake-plugin',
+    recordServices: [fakeService]
+  };
+
+  plugins.getRecordingPlugins = sinon.fake.returns([fakePlugin]);
+
+  recordingHistory.set('activeRecording', {
+    filePath: incomplete,
+    name: 'Incomplete',
+    date: new Date().toISOString(),
+    apertureOptions: {},
+    plugins: {
+      'kap-fake-plugin': {
+        'Fake Service': {
+          some: 'state'
+        }
+      }
+    }
+  });
+
+  const checkPromise = checkForActiveRecording();
+
+  const dialogState = await dialog.waitForDialog();
+  t.true(dialogState.detail.includes('playable'));
+
+  // Don't delete it until user is done interacting with the dialog
+  t.true(recordingHistory.has('activeRecording'));
+
+  await dialog.fakeAction(dialogState.buttons.findIndex(b => b.label && b.label.toLowerCase().includes('editor')));
+  t.deepEqual(openEditorWindow.lastCall.args, [
+    incomplete,
+    {recordingName: 'Incomplete'}
+  ]);
+
+  t.true(await checkPromise);
+
+  t.false(recordingHistory.has('activeRecording'));
+  t.true(fakeService.cleanUp.calledOnceWith({some: 'state'}));
+  t.deepEqual(recordingHistory.get('recordings'), [
+    {filePath: incomplete, name: 'Incomplete', date: new Date().toISOString()}
+  ]);
```

Comment:
While this is more verbose, it's also IMHO more readable, as each assertion value are on separate lines.

```suggestion
  t.deepEqual(
    recordingHistory.get('recordings'),
    [
      {
        filePath: incomplete, name: 'Incomplete', date: new Date().toISOString()
      }
    ]
  );
```